skimage/segmentation/manual_segmentation.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.patches import Polygon
+from matplotlib.collections import PatchCollection
+
+
+def manual(image, speed_up=1):
+    """Return a binary image based on the selections made with mouse clicks.
+
+    Parameters
+    ----------
+    image : (M, 2) array
+        Grayscale or RGB 2D image.
+
+    speed_up : int
+        Skips vertices in integer steps to speed up mask generation. This
+        must be a non-zero integer.
+
+    Returns
+    -------
+    mask : (M, 2) binary array
+        Segmented binary mask.
+
+    Notes
+    -----
+    Use the cursor to draw objects. Increasing speed_up value will 
+    generate the mask faster, but the objects will be less smoother. 
+
+    Examples
+    --------
+    >>> from skimage import data, segmentation
+    >>> camera = data.camera()
+    >>> mask = segmentation.manual(camera)
+    >>> # Use the cursor to draw objects
+    >>> io.imshow(mask)
+    >>> io.show()
+
+    """
+    image = np.squeeze(image)
+
+    if image.ndim not in (2, 3):
+        raise TypeError('Only 2-D images or 3-D images supported.')
+
+    manual.list_of_verts = []
+    manual.polygons_selection = []
+    manual.polygon_list = []
+
+    _select_lasso(image)
+
+    if image.ndim is 3:
+        image = np.squeeze(image[:, :, :1])
+
+    mask = np.zeros(image.shape)
+
+    yshape, xshape = image.shape
+    y_grid, x_grid = np.mgrid[:yshape, :xshape]
+    all_pixels = np.vstack((x_grid.ravel(), y_grid.ravel())).T
+
+    for verts in manual.list_of_verts:
+        verts_ = verts[::speed_up]
+        _path = matplotlib.path.Path(verts_, closed=True)
+        _mask = _path.contains_points(all_pixels)
+        _mask = _mask.reshape(image.shape)
+        mask += _mask
+    return mask >= 1
+
+
+def _select_lasso(image):
+    """Uses the LassoSelector widget from matplotlib
+    to crop freehand."""
+    fig, _select_lasso.ax = plt.subplots()
+    _select_lasso.ax.imshow(image)
+
+    buttonpos = plt.axes([0.85, 0.5, 0.1, 0.075])
+    undo_button = matplotlib.widgets.Button(buttonpos, "Undo")
+    undo_button.on_clicked(_undo)
+
+    lasso = matplotlib.widgets.LassoSelector(_select_lasso.ax, _onselect)
```

Comment:
Here, the way you've implemented it, `_on_select` needs to refer to a "global" attribute, `manual.list_of_verts`. Instead, you can create a nested [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)), just above:

```python
list_of_verts = []
def _on_select(verts):
    list_of_verts.append(verts)
    # ...
lasso = matplotlib.widgets.LassoSelector(_select_lasso.ax, _onselect)
return list_of_verts
```